src/storage.ts

```diff
@@ -25,16 +40,36 @@ const V0_DEDUP_RTV_PREFIXES: ReadonlySet<string> = new Set([
  * https://developers.cloudflare.com/workers/learning/how-the-cache-works#interacting-with-the-cloudflare-cache
  *
  * @param url - to fetch from cache or network.
+ * @param cf - incoming Cloudflare properties.
  * @returns a Response object for the request URL.
  */
-export async function fetchImmutableUrlOrDie(url: string): Promise<Response> {
+export async function fetchImmutableUrlOrDie(
+  url: string,
+  cf?: IncomingCloudflarePropertiesExtended
+): Promise<Response> {
   const response = await fetch(url, {
     cf: {cacheEverything: true, cacheTtl: 31536000},
   });
   if (!response.ok) {
     throw new FetchError(response.status, response.statusText);
   }
-  return response;
+
+  if (!tryBrotli(cf)) {
+    return response;
+  }
+
+  const brotliResponse = await fetch(`${url}.br`, {
+    cf: {cacheEverything: true, cacheTtl: 31536000},
+  });
+  if (!brotliResponse.ok) {
+    console.warn('Brotli pre-compressed file for', url, 'not found in storage');
+    return response;
+  }
+
+  // Merge the contents of Brotli response with headers from raw response.
+  const mergedResponse = new Response(brotliResponse.body, response);
+  mergedResponse.headers.set(HeaderKeys.CONTENT_ENCODING, 'br');
+  return mergedResponse;
```

Comment:
We also need to se `encodeBody: 'manual'` according to https://developers.cloudflare.com/workers/runtime-apis/response#properties

Reply:
Fixed 35574b5

Reply:
Fixed 35574b5